tests/features/test_covered_distance.py

```diff
@@ -0,0 +1,194 @@
+                    np.sqrt(2), 
+                    np.sqrt(2), 
+                    np.sqrt(10), 
+                    np.sqrt(5)
+                ],
+                "away_2_ax": [1, -0.5, -1, 1, 0.5, -1],
+                "away_2_ay": [0, 0, 0, 0, 0, 0],
+                "away_2_acceleration": [
+                    1, 
+                    0.5, 
+                    1, 
+                    1, 
+                    0.5, 
+                    1]
+            }
+        )
+
+        self.expected_total_distance = {
+            'home_1': {
+                'total_distance': (np.sqrt(149) + np.sqrt(556.25) + np.sqrt(181) + np.sqrt(325))
+            },
+            'away_2': {
+                'total_distance': (np.sqrt(5) + np.sqrt(10) + np.sqrt(2) + np.sqrt(2) + np.sqrt(10) + np.sqrt(5))
+                }
+        }
+
+        self.expected_output_total_and_vel_distance = {
+            'home_1': {
+                'total_distance': (np.sqrt(149) + np.sqrt(556.25) + np.sqrt(181) + np.sqrt(325)),
+                'total_distance_velocity': [((-1, 13), np.sqrt(149)),((17, 30), (np.sqrt(556.25) + np.sqrt(325)))] 
+            },
+            'away_2': {
+                'total_distance': (np.sqrt(5) + np.sqrt(10) + np.sqrt(2) + np.sqrt(2) + np.sqrt(10) + np.sqrt(5)), 
+                'total_distance_velocity': [((-1, 13), (np.sqrt(5) + np.sqrt(10) + np.sqrt(2) + np.sqrt(2) + np.sqrt(10) + np.sqrt(5))),((17, 30), 0)] 
+            }
+        }
+
+        self.expected_output_total_and_vel_and_acc_distance = {
+            'home_1': {
+                'total_distance': (np.sqrt(149) + np.sqrt(556.25) + np.sqrt(181) + np.sqrt(325)),
+                'total_distance_velocity': [((-1, 13), np.sqrt(149)), ((17, 30), (np.sqrt(556.25) + np.sqrt(325)))], 
+                'total_distance_acceleration': [((15, 37), np.sqrt(149)),((0, 0.75), 0)]
+            },
+            'away_2': {
+                'total_distance': (np.sqrt(5) + np.sqrt(10) + np.sqrt(2) + np.sqrt(2) + np.sqrt(10) + np.sqrt(5)), 
+                'total_distance_velocity': [((-1, 13), (np.sqrt(5) + np.sqrt(10) + np.sqrt(2) + np.sqrt(2) + np.sqrt(10) + np.sqrt(5))), ((17, 30), 0)] , 
+                'total_distance_acceleration': [((15, 37), 0), ((0, 0.75), (np.sqrt(10) + np.sqrt(10)))]
+            }
+        }
+
+    # test covered distance
+    def test_get_total_distance(self):
+        input_df = self.input.copy()
+        input_df.drop(columns=["home_1_vx"], inplace=True)
+        with self.assertRaises(ValueError):
+            get_covered_distance(input_df, ["home_1","away_2"], self.framerate)
+
+        input_df = self.input.copy()
+        output = get_covered_distance(input_df, ["home_1","away_2"], self.framerate)
+        self.assertDictEqual(output, self.expected_total_distance)
+
+    # test covered distance and velocity interval
+    def test_get_total_and_vel_distance(self):
+        input_df = self.input.copy()
+        vel_intervals = self.vel_intervals
```

Comment:
Uneccesarry variable, just add `self.vel_intervals` directly in the function call (also check the same logic in other tests)